File: sudoku-auth/auth-boot/src/main/java/com/zumin/sudoku/auth/controller/PublicKeyController.java
package com.zumin.sudoku.auth.controller;

import com.nimbusds.jose.jwk.JWKSet;
import com.nimbusds.jose.jwk.RSAKey;
import com.zumin.sudoku.common.web.annotation.ComRestController;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import java.security.KeyPair;
import java.security.interfaces.RSAPublicKey;
import java.util.Map;
import lombok.AllArgsConstructor;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@ComRestController(tags = "获取公钥接口")
public class PublicKeyController {

  private final KeyPair keyPair;

  @ApiOperation("获取公钥")
  @GetMapping("/getPublicKey")
  public Map<String, Object> loadPublicKey() {
    RSAPublicKey publicKey = (RSAPublicKey) keyPair.getPublic();
    RSAKey key = new RSAKey.Builder(publicKey).build();
    return new JWKSet(key).toJSONObject();
  }

}

File: sudoku-auth/auth-boot/src/main/java/com/zumin/sudoku/auth/controller/PublicKeyController.kt
package com.zumin.sudoku.auth.controller

import com.nimbusds.jose.jwk.JWKSet
import com.nimbusds.jose.jwk.RSAKey
import com.zumin.sudoku.common.web.ComRestController
import io.swagger.annotations.ApiOperation
import org.springframework.web.bind.annotation.GetMapping
import java.security.KeyPair
import java.security.interfaces.RSAPublicKey

@ComRestController(tags = ["获取公钥接口"])
class PublicKeyController(private val keyPair: KeyPair) {

  @ApiOperation("获取公钥")
  @GetMapping("/getPublicKey")
  fun loadPublicKey(): Map<String, Any> {
    val publicKey = keyPair.public as RSAPublicKey
    val key = RSAKey.Builder(publicKey).build()
    return JWKSet(key).toJSONObject()
  }
}